ecosystem/sep-0010.md

```diff
@@ -165,14 +175,16 @@ To validate the challenge transaction the following steps are performed by the *
   * is a Manage Data operation
   * has a non-null source account
 * verify that transaction envelope has a correct signature by the **Server Account**
-* if the operation's source account exists:
+* if the first operation's source account exists:
   * verify that the remaining signature count is one or more;
-  * verify that remaining signatures on the transaction are signers of the operation's source account;
+  * verify that remaining signatures on the transaction are signers of the **Client Account**
   * verify that remaining signatures are correct;
   * verify that remaining signatures provide weight that meets the required threshold(s), if any;
-* if the operation's source account does not exist:
+* if the first operation's source account does not exist:
   * verify that remaining signature count is one;
   * verify that remaining signature is correct for the master key of the **Client Account**
+* if the transaction contains a Manage Data operation with the key `client_domain`:
+  * verify that the transaction was signed by the source account of the Manage Data operation
```

Comment:
I know this is further described in the doc (in the "Verifying the Client Domain" section), but I'd still add these extra steps here for clarity:

```suggestion
  * verify that the `value` provided in this Manage Data operation is a domain that serves a valid [SEP-1 stellar.toml](sep-0001.md) file containing a `SIGNING_KEY` attribute (see [Verifying the Client Domain](#verifying-the-client-domain))
  * verify that the source account of this Manage Data operation is the same as the `SIGNING_KEY` retrieved from the `stellar.toml`
  * verify that the transaction was signed by the source account of this Manage Data operation
```

Reply:
The server doesn't need to check that the `SIGNING_KEY` matches the source account in the `POST` request logic because the server assigns `SIGNING_KEY` as the source account in the `GET` request logic and signs the transaction.

The source account cannot be altered without invalidating the server's signature, so the server can trust it's past work by verifying it's own signature.

Reply:
Oh yeah you're right, the server already sends a signed transaction on `GET`, so verifying its own signature should be enough in that case.